azure-pipelines.yaml

```diff
@@ -0,0 +1,162 @@
+      matrix:
+        mac_36:
+          imageName: 'macOS-10.14'
+          python.version: '3.6'
+        mac_37:
+          imageName: 'macOS-10.14'
+          python.version: '3.7'
+        linux_py36:
+          imageName: 'ubuntu-16.04'
+          python.version: '3.6'
+        linux_py37:
+          imageName: 'ubuntu-16.04'
+          python.version: '3.7'
+    pool:
+      vmImage: $(imageName)
+
+    steps:
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(python.version)'
+        architecture: 'x64'
+
+    - script: |
+        brew uninstall node
+        brew unlink node@6
+        brew update
+        brew upgrade
+        brew cleanup
+        brew install yarn
+        brew install rbenv/tap/openssl@1.0
+        brew switch openssl 1.0.2t
+      displayName: 'Install NodeJS and yarn'
+      condition: eq(variables['imageName'], 'macOS-10.14')
+
+    - script: |
+        sudo apt-get install curl
+        curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
+        sudo apt-get install nodejs
+      displayName: 'Install NodeJS'
+      condition: eq(variables['imageName'], 'ubuntu-16.04')
+
+    - script: |
+        curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+        echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+        sudo apt update && sudo apt install yarn
+      displayName: 'Install yarn'
+      condition: eq(variables['imageName'], 'ubuntu-16.04')
+
+    - script: python -m pip install --upgrade ipywidgets traitlets pytest wheel jupyter_packaging
+      # pip installs release candidates by default:
+      #   https://github.com/pypa/pip/issues/4969
+      displayName: 'Install dependencies'
+
+    - script: python -m pip install --upgrade black
+      displayName: 'Install black (linter)'
+      condition: eq(variables['python.version'], '3.7')
+    - script: black --check .
+      displayName: 'Test Formating'
+      condition: eq(variables['python.version'], '3.7')
+
+    - script: python -m pip install --upgrade .
+      displayName: 'Test Install'
+      condition: ne(variables['Agent.OS'], 'Windows_NT')
+      env:
+        NPM_TOKEN: $(NPM_TOKEN)
+
+      # Build the python distribution from source
+    - script: python setup.py bdist_wheel
+      displayName: 'Build wheel'
+      condition: ne(variables['Agent.OS'], 'Windows_NT')
+      env:
+        NPM_TOKEN: $(NPM_TOKEN)
+
+    - bash: |
+        set -xeo pipefail
+        # Display log files if the build failed
+        echo "Dumping log files for failed build"
+        echo "----------------------------------"
+        for f in $(find $BUILD_REPOSITORY_LOCALPATH/build -name *.log);
+          do echo "------"
+              echo $f
+              echo "======"
+              cat $f
+          done;
+      condition: failed() # only run this job if the build step failed
+      displayName: "Print log files (failed build only)"
+- stage: Deploy
```

Comment:
You can add a condition on the stage for `condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))` instead of having the condition of the ref tag on each step.

Reply:
@Shelnutt2 ty for noticing! I will check on that.